pkg/controller/download_controller.go

```diff
@@ -297,6 +298,34 @@ func (c *downloadController) patchDownload(req *pluginv1api.Download, mutate fun
 	return req, nil
 }
 
+func (c *downloadController) patchDownloadByStatusWithRetry(req *pluginv1api.Download, newPhase pluginv1api.DownloadPhase, msg string) (*pluginv1api.Download, error) {
+	var updatedDownload *pluginv1api.Download
+	var err error
+	var retryCnt = 0
+	log := loggerForDownload(c.logger, req)
+	log.Info("Ready to call patchDownloadByStatus API. Will retry on network issue for %d times.", utils.RetryMaximum)
```

Comment:
`Will retry on network issue` --> `Will retry on patch failure of Download status every XXX seconds up to XXX seconds`